### PR TITLE
Rewrite introduction

### PR DESCRIPTION
**Pull request by heeckhau**

### CODE REVIEW

File: src/intro.md [hunk 1]
```diff
@@ -1,35 +1,34 @@
...
+<!-- - Transport Layer Security (TLS)
     - Encryption: hides data from third parties
     - Authentication: ensures that the parties exchanging information are who they claim to be
-    - Integrity: verifies that data has not been forged or tampered with]: #
-
+    - Integrity: verifies that data has not been forged or tampered with -->
 ### ② Notarization
 
-The `User` forwards the encrypted TLS traffic to the `Notary`, which checks that it has not been tampered with and notarizes the whole TLS session by signing a transcript of it.
+The `User` forwards the encrypted TLS traffic to the `Notary`, which validates its integrity and then notarizes the entire TLS session by signing its transcript.
```
Comment (themighty1) [inline src/intro.md:31]: In light of what I suggested above, Id rephrase (rough draft without careful word choice)

The `User` engages in MPC with the `Notary` to perform all cryptographic operations on the TLS traffic. Thanks to the properties of MPC, the Notary never learns the data which he signs and yet he is able to validate the authenticity of the data.

**Review by themighty1 — APPROVED** [review]
gw

File: src/intro.md [hunk 2]
```diff
@@ -1,35 +1,34 @@
...
+TLSNotary works by adding a third party, the `Notary`, to the usual TLS connection between the `User` and a Server. This `Notary` is **not an intermediary**. Instead, the `Notary` participates in a **secure multi-party computation** (MPC) to jointly operate the TLS connection without ever seeing the data in plain text; the `Notary` only sees encrypted data. Given that the `Notary` only sees the ephemeral keys of the Server, the `Notary` does not know which Server the `User` is communicating with. The TLSNotary protocol is transparent to the Server. From the Server's perspective, the TLS connection is the same as all other connections. Therefore, no modifications to the TLS protocol are needed.
```
[the remark this answers is not in the document]
Comment (heeckhau) [reply src/intro.md:23]: I'll replace "intermediary" with "a man in the middle", but leave the rest as it is for now. The intro's intention is to explain the general concepts and emphasise that the protocol will not leak private information.

[the remark this answers is not in the document]
Comment (heeckhau) [reply src/intro.md:23]: I think it is a good idea to repeat a few times that the protocol respects the User's privacy.

[the remark this answers is not in the document]
Comment (heeckhau) [reply src/intro.md:23]: ✅ 
I'll also simplify the language and replace it with "temporary" key

[the remark this answers is not in the document]
Comment (heeckhau) [reply src/intro.md:23]: I'll drop this sentence

File: src/intro.md [hunk 3]
```diff
@@ -1,35 +1,62 @@
...
+    - Encryption: hides data from third parties
+    - Authentication: ensures that the parties exchanging information are who they claim to be
+    - Integrity: verifies that data has not been forged or tampered with -->
+### ② Notarization
+
+The `User` forwards the encrypted TLS traffic to the `Notary`, which validates its integrity and then notarizes the entire TLS session by signing its transcript.
+
+### ③ Selective Disclosure
+
+The TLSNotary protocol offers a way for the `User` to selectively prove the authenticity of arbitrary portions of the transcript to a Verifier. Because of the way that the `Notary` signs the transcript, the `User` can prove the authenticity of the transcript to any third-party `Verifier` using Zero Knowledge Proofs.
```
Comment (th4s) [inline src/intro.md:35]: My understanding here is a little bit different.
Selective disclosure means that the user can redact parts of the transcript when showing it to the verifier. There are no ZK proofs involved.

ZK proofs come into play when you want to prove something in ZK about redacted parts, like e.g. "the age field is redacted, but here is the ZK proof that it is over 18".

File: src/intro.md (hunk 1 above)
Comment (heeckhau) [reply src/intro.md:31]: Thanks for comments. I have rephrased this entire paragraph.

File: src/intro.md [hunk 4]
```diff
@@ -1,35 +1,62 @@
...
+
+### ④ Verification
+
+A `Verifier` can verify the proof they receive from the `User`. In this context, the `User` is often called the `Prover`, because they offer the proof. Thanks to the zero-knowledge proofs, the `User` can prove the requirement statement without disclosing any extra information, thus protecting their privacy.
```
Comment (th4s) [inline src/intro.md:39]: ZK proofs are not necessary. They only come into play when proving statements about redacted parts.

File: src/intro.md [hunk 5]
```diff
@@ -1,35 +1,62 @@
...
+
+### Trust Assumptions
+
+TLSNotary requires  trust assumption. A `Verifier` of a proof must trust that the `Notary` did not collude with the `User` to forge it. This trust can be minimized by requiring multiple proofs, each signed by different Notaries.
```
Comment (th4s) [inline src/intro.md:43]: ..requires a trust assumption.

**Review by th4s — COMMENTED** [review]
Good work :rocket:

File: src/intro.md [hunk 6]
```diff
@@ -1,35 +1,67 @@
...
+1. The `User` **requests** the data from the `Server` privately and securely.
+2. The `Notary` **checks** for data tampering and **signs** the data.
+3. The `User` **selectively discloses** the data to the `Verifier`.
+4. The `Verifier` **verifies** the data.
+
+![](./png-diagrams/overview3.png)
+
+### ① Multi-party TLS Request
+
+TLSNotary works by adding a third party, the `Notary`, to the usual TLS connection between the `User` and a `Server`. This `Notary` is **not "a man in the middle"**. Instead, the `Notary` participates in a **secure multi-party computation** (MPC) to jointly operate the TLS connection without ever seeing the data in plain text; the `Notary` only sees encrypted data. Given that the `Notary` only sees the temporary key of the `Server`, the `Notary` does not know which `Server` the `User` is communicating with. The TLSNotary protocol is transparent to the `Server`. From the `Server`'s perspective, the User's connection is a standard TLS connection.
```
[the remark this answers is not in the document]
Comment (heeckhau) [reply src/intro.md:23]: 👍
I'll add the link, but will not go into this implementation detail here.
We can explain this in the browser extension later

**Pull request merged**